File: Problems/No.12_Integer_to_Roman/integer_to_roman.cpp
class Solution {
public:
    string intToRoman(int num) {
        
        // create the list of integer-to-roman
        int integer[] = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
        string rom[] = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
        string ItR_ans = "";
        
        // start compute
        for (int i=0; i<sizeof(integer)/sizeof(*integer); i++) {
            
            while (integer[i] <= num) {
                
                num -= integer[i];
                ItR_ans += rom[i];
                
            }
            
        }
        
        return ItR_ans;
        
    }
};

File: Problems/No.12_Integer_to_Roman/integer_to_roman.py
class Solution(object):
    def intToRoman(self, num):
        """
        :type num: int
        :rtype: str
        """
        
        # the mapping of integer to roman
        int_to_roman = [
            [1000, "M"],
            [900, "CM"],
            [500, "D"],
            [400, "CD"],
            [100, "C"],
            [90, "XC"],
            [50, "L"],
            [40, "XL"],
            [10, "X"],
            [9, "IX"],
            [5, "V"],
            [4, "IV"],
            [1, "I"]
        ]
        ItR_ans = ""
        
        # start compute
        for integer, roman in int_to_roman:
            while (num >= integer):
                num -= integer
                ItR_ans += roman
                
        return ItR_ans
